tests/lookup/tests.py

```diff
@@ -1527,3 +1530,21 @@ def test_conditional_expression(self):
                 {"year": 2042, "century": "other"},
             ],
         )
+
+    def test_multivalued_join_reuse(self):
+        self.assertEqual(
+            Season.objects.get(Exact(F("games__home"), "NY"), games__away="Boston"),
+            self.s1,
+        )
+        self.assertEqual(
+            Season.objects.get(Exact(F("games__home"), "NY") & Q(games__away="Boston")),
+            self.s1,
+        )
```

Comment:
Let’s make sure changing the order doesn’t affect anything the way it happened in the ticket
```suggestion
        self.assertEqual(
            Season.objects.get(Exact(F("games__home"), "NY") & Q(games__away="Boston")),
            self.s1,
        )
        self.assertEqual(
            Season.objects.get(Q(games__away="Boston") & Exact(F("games__home"), "NY")),
            self.s1,
        )
```

Reply:
Ditto